common/perception_utils/test/src/test_object_classification.cpp

```diff
@@ -61,3 +61,114 @@ TEST(object_classification, test_getHighestProbLabel)
+    EXPECT_TRUE(isVehicle(ObjectClassification::CAR));
+    EXPECT_TRUE(isVehicle(ObjectClassification::MOTORCYCLE));
+    EXPECT_TRUE(isVehicle(ObjectClassification::TRAILER));
+    EXPECT_TRUE(isVehicle(ObjectClassification::TRUCK));
+  }
+
+  // False Case with uint8_t
+  {
+    EXPECT_FALSE(isVehicle(ObjectClassification::UNKNOWN));
+    EXPECT_FALSE(isVehicle(ObjectClassification::PEDESTRIAN));
+  }
+
+  // True Case with object_classifications
+  {  // normal case
+    std::vector<autoware_auto_perception_msgs::msg::ObjectClassification> classification;
+    classification.push_back(createObjectClassification(ObjectClassification::CAR, 0.5));
+    classification.push_back(createObjectClassification(ObjectClassification::TRUCK, 0.8));
+    classification.push_back(createObjectClassification(ObjectClassification::BUS, 0.7));
+    EXPECT_TRUE(isVehicle(classification));
+  }
+
+  // False Case with object_classifications
+  {  // false case
+    std::vector<autoware_auto_perception_msgs::msg::ObjectClassification> classification;
+    classification.push_back(createObjectClassification(ObjectClassification::PEDESTRIAN, 0.8));
+    classification.push_back(createObjectClassification(ObjectClassification::BICYCLE, 0.7));
+    EXPECT_FALSE(isVehicle(classification));
+  }
+
+}  // TEST isVehicle
+
+// TEST isCarLikeVehicle
+TEST(object_classification, test_isCarLikeVehicle)
+{
+  using autoware_auto_perception_msgs::msg::ObjectClassification;
+  using perception_utils::isCarLikeVehicle;
+
+  {  // True Case with uint8_t
+    EXPECT_TRUE(isCarLikeVehicle(ObjectClassification::BUS));
+    EXPECT_TRUE(isCarLikeVehicle(ObjectClassification::CAR));
+    EXPECT_TRUE(isCarLikeVehicle(ObjectClassification::TRAILER));
+    EXPECT_TRUE(isCarLikeVehicle(ObjectClassification::TRUCK));
+  }
+
+  // False Case with uint8_t
+  {
+    EXPECT_FALSE(isCarLikeVehicle(ObjectClassification::UNKNOWN));
+    EXPECT_FALSE(isCarLikeVehicle(ObjectClassification::BICYCLE));
+    EXPECT_FALSE(isCarLikeVehicle(ObjectClassification::PEDESTRIAN));
+    EXPECT_FALSE(isCarLikeVehicle(ObjectClassification::MOTORCYCLE));
+  }
+
+  // True Case with object_classifications
+  {  // normal case
+    std::vector<autoware_auto_perception_msgs::msg::ObjectClassification> classification;
+    classification.push_back(createObjectClassification(ObjectClassification::CAR, 0.5));
+    classification.push_back(createObjectClassification(ObjectClassification::TRUCK, 0.8));
+    classification.push_back(createObjectClassification(ObjectClassification::BICYCLE, 0.7));
+    EXPECT_TRUE(isCarLikeVehicle(classification));
+  }
+
+  // False Case with object_classifications
+  {  // false case
+    std::vector<autoware_auto_perception_msgs::msg::ObjectClassification> classification;
+    classification.push_back(createObjectClassification(ObjectClassification::MOTORCYCLE, 0.8));
+    classification.push_back(createObjectClassification(ObjectClassification::BICYCLE, 0.8));
```

Comment:
How about adding a test to see what will happen if the object classification includes car-like vehicle and not car-like vehicle with the same probability.
This will be the edge case for isCarLikeVehicle function.

Reply:
That makes sense. I'll do it.

Reply:
This edge case depends following function `getHighestProbLabel`, which extract highest-scored label from classification.

- This function do not considering multiple highest probability
- It returns an earlier label with maximum probability. (**It depends labels order!**)
- For example, if an objects has CAR and MOTORCYCLE label with the same probability,
  - `isCarLikeVehicle` return `true` when CAR label is added first
  - else it returns `false`

```cpp
inline std::uint8_t getHighestProbLabel(
  const std::vector<ObjectClassification> & object_classifications)
{
  std::uint8_t label = ObjectClassification::UNKNOWN;
  float highest_prob = 0.0;
  // TODO(Satoshi Tanaka): It might be simple if you use STL or range-v3.
  for (const auto & object_classification : object_classifications) {
    if (highest_prob < object_classification.probability) {
      highest_prob = object_classification.probability;
      label = object_classification.label;
    }
  }
  return label;
}
```

Reply:
Got it. Even if the current implementation is not good, in order to show how it works with the edge case in the current implementation, adding this edge-case test seems useful.
What do you think?

Reply:
Frankly, I think it is a rare situation that the label probabilities become the same since it is float value.
But adding edge case seems to be not harmful and it is easy to implement, I added some edge cases to this PR.